### PR TITLE
Added missing unescaping in named collections names. 

### PR DESCRIPTION
**Pull request by MikhailBurdukov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added missing unescaping in named collections. Without fix clickhouse-server can't start.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

Also can we make backport of the pr?

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 936d6b22518e7711adc4991663f6474b42805eb8 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71308/936d6b22518e7711adc4991663f6474b42805eb8/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/e29c488b05a3026a8c4b502dd7e111357acae982/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by MikhailBurdukov** [comment]
Have restarted the CI because it was a lot of tests failures that looks unrelated with changes for me

File: tests/integration/test_named_collections/test.py [hunk 1]
```diff
@@ -794,3 +794,17 @@ def test_keeper_storage_remove_on_cluster(cluster, ignore, expected_raise):
...
+
+
+@pytest.mark.parametrize(
+    "instance_name",
+    [("node"), ("node_with_keeper")],
+)
+def test_name_escaping(cluster, instance_name):
+    node = cluster.instances[instance_name]
+
+    node.query("DROP NAMED COLLECTION IF EXISTS test;")
```
Comment (kssenii) [inline tests/integration/test_named_collections/test.py:806]:
```suggestion
    node.query("DROP NAMED COLLECTION IF EXISTS `test_!strange/symbols!`;")
```

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> Upgrade check (debug) — New settings are not reflected in settings changes history 

query_plan_join_inner_table_selection -- not from this PR

> AST fuzzer (ubsan)

https://github.com/ClickHouse/ClickHouse/issues/71454

> Stateless tests (debug, s3 storage) — fail: 1

01290_max_execution_speed_distributed -- not related to changes

**Comment by MikhailBurdukov** [comment]
@kssenii Can we also make a backport?

**Comment by kssenii** [comment]
Ok

**Pull request merged**